### ISSUE TITLE
[Beta23] Odd Flux Rift Behavior

### ISSUE THREAD

**Issue by apmax** [Azanor/thaumcraft-beta#1353]
The flux event that spawns taint seeds is bizarrely rare.
I've been trying to get taint to spawn in survival with Beta22 for a while now, and unless a taint seed is intended to be an **extremely** rare event now, something feels broken.  I've attempted to grow and feed several rifts until one spawns, but every attempt to do so has only resulted in the rift inexplicably and randomly exploding into void seeds as if I'd collapsed it.   This behavior persists into Beta23.
Is it intended behavior that flux rifts will randomly explode into void seeds for free more often than they will attempt to spawn a taint seed?

**Comment by lOmicronl** [comment]
Lucky you. I've been trying to get void seeds without any luck. Any rift I spawn and watch close itself results in a taint seed and zero drops. Literally zero, no void seeds, no pearls, nothing.

Can you share your process _exactly_? Step by step? I'll offer mine for comparison. Perhaps each of us is doing things a particular way that changes the way the rifts respond?

My process:

* Set up crucible on top of burning netherrack in the middle of a chunk, an everfull urn close by
* Drop full stack of cobblestone into crucible
* Shift-click with an empty hand to dump crucible contents into the aura
* Use my caster to rapidly deplete the local aura
* Usually this is enough to spawn the rift; if not, I give it another stack of cobble.
* Dump more cobble into the crucible and leave it there, letting it evaporate on its own, thereby releasing flux at a constant rate for the rift to feed on
* Do this until rift has reached a nice size - usually 7-8 blocks in length
* Excavate the pit the rift naturally dug a bit more, to be able to place several hungry chests at the bottom, for catching any drops
* Go AFK in a nearby save space for an hour or two, reading a book or watching youtube, while occasionally peering at the screen
* Investigate as soon as sounds indicate that the rift has closed

For me, this results in 100% taint seed spawn and 0% drops of any kind.

**Comment by apmax** [comment]
My method is just that I brought several full warded jars to the chunk, and tried to keep the amount of flux consistently at 800 or so.  Checking back every once in a while when the spike on my dioptra spread out noticeably.  Every time I've done this, a point has come where I hear explosions and see the rift in the distance close and void seeds pop out of it.  On my first attempt I tried keeping it at 2000 flux, but the rift eventually simply exploded into void seeds and the flux dropped back to 100 or so, as it has on every further attempt.

Overflowing a chunk with flux doesn't seem to help much either way, all it does is make it so that another rift will spawn almost as soon as the present one closes.  
I would think that more flux in a chunk with a rift means it's more likely for big and dangerous flux events to happen, like taint seeds spawning, but I wonder if flux events might be truly random: independent of total flux content in the chunk the rift is in, just needing a certain minimum flux content.
I don't know if I have any useful balance suggestions but I do think rifts need a bit it.

**Comment by lOmicronl** [comment]
Well, one big difference is definitely the way you have way more flux up than I do. I generally have 300-400 flux at the beginning, then the rift spawns and it drops rapidly. The evaporation from the crucible is actually slightly slower than the consumption abilities of the rift, so the flux will be at a low level throughout the process.

I can certainly try to see if I get a different result if I force the chunk to be full of flux all the time. Maybe there is a maximum rift size, and once it exceeds this it automatically explodes?

**Comment by apmax** [comment]
Oh, there is a maximum rift size, but it does not immediately trigger an explosion, and it requires WAY more flux than I'm pumping out.  Go into a creative world and dump a few dozen stacks or so of alchemical furnaces into a crucible and dump it all out at once.  Max rift size seems to be way larger than before, terrifyingly so.  I think it'd be kind of cool to have semi-stable rifts that big as rare world-gen features, they're quite a spectacle.   Destroying a max-sized rift with a causality collapser nets around 17 void seeds.  The natural explosion of tiny rifts nets something around 7 seeds.

My rifts sit there with a decent amount of flux around for anywhere from 30 minutes to two hours before they explode and vanish.   I had a low-flux rift inside my base for an excessive amount of time as I progressed and not a single taint seed appeared.

**Comment by lOmicronl** [comment]
Right, so it turns out you were probably just unlucky.

Or, rather: you were _absurdly lucky_.

Someone on reddit looked into the .jar, and said that there are two distinct possibilities for a rift to cease existing. One, the rift spawns a Taint seed and vanishes. Two, the rift collapses explosively. _Only_ the collapsing mechanic results in item drops of any kind. However, if you do not force it via a Causality Collapser, only about one in six rifts will spontaneously collapse. Five out of six will naturally result in Taint.

In other words: you constantly keep rolling below 17 on a d100. I constantly keep rolling above.

**Comment by apmax** [comment]
Huh.  I guess it's expected with actual RNG, but still.  I don't know if I'm a fan of purely RNG flux events like that.  It would make more sense for a higher flux content to increase the probability of bad events like taint, and low flux content to be the only condition under which a rift could lose stability and explode naturally, but those ideas are probably more in the realm of suggestion than bug.

Gameplay balance around flux rifts and flux events seems to be still in flux, anyway, though, I wonder how many other people are encountering weird strings of misplaced luck.

**Comment by Azanor** [comment]
I have tweaked the numbers in B25 a bit making taint more likely, but a natural collapse should be very rare - roughly 1 in 100 chance. A taint seed event could occur about 1 in 16 times as stated above. This is changing to 1 in 10 in the B25.
